### PR TITLE
Added: Go to Implementation

### PR DESCRIPTION
**Pull request by eidheim**
Works if implementation file is opened and parsed. Header files are not considered, use Go to Declaration instead then. Still pretty useful.

### CODE REVIEW

**Comment by eidheim** [comment]
@zalox would you mind have a quick look at this? I'll probably cleanup the refactoring methods at some point later on.

**Comment by eidheim** [comment]
Also, lets say I want to combine these 3 commits into one, without opening another pull request, how would I do that?

**Comment by eidheim** [comment]
@zalox I think I managed to merge the 3 commits into 1, but double check it for me please:)

File: src/source_clang.cc [hunk 1]
```diff
@@ -919,6 +919,29 @@ Source::ClangViewAutocomplete(file_path, language) {
...
+           a_token.get_kind()==clang::Token_Identifier && cursor.has_type()) {
+          auto referenced=cursor.get_referenced();
+          if(referenced && static_cast<clang::CursorKind>(token.type)==referenced.get_kind() &&
+             token.spelling==a_token.get_spelling() && token.usr==referenced.get_usr()) {
+            location.file_path=cursor.get_source_location().get_path();
+            auto clang_offset=cursor.get_source_location().get_offset();
+            location.line=clang_offset.line;
+            location.index=clang_offset.index;
+            //do not break here, choose last occurrence in case declaration and implementation is in the same file
+          }
```
Comment (zalox) [inline src/source_clang.cc]: Why not iterate backwards?

Comment (eidheim) [inline src/source_clang.cc]: Good idea!

Comment (zalox) [inline src/source_clang.cc]: It might be a good idea to communicate to the user why a jump fails when the implementation file is closed.

Comment (eidheim) [inline src/source_clang.cc]: Yeah, at some point later we should add message boxes, for instance top left of the window, that disappears after a few seconds. There might be support for this in gtk already, will investigate in the summer.

**Comment by eidheim** [comment]
Reversed iteration fixed, also now looks for non-class function implementations. I'll combine the last commit together with the first after you have looked through the changes.

**Comment by zalox** [comment]
Looks good!

The commits are fine, they are descriptive and does what they tell us. I only want to avoid those "fixes to last commit.", because they interrupt workflow when iterating through changes. I think it's more important to maybe be more careful about pushes, and therefore consider doing a `git rebase -i` if there are git-cleanup needed since the last push. I don't want to promote `--force` pushing since that can be extremely dangerous.

**Comment by eidheim** [comment]
Sounds good, merging then. Added smaller/temporary juCi++ messages (not in terminal) to my TODO.

**Pull request merged**